contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -111,16 +91,24 @@ abstract contract ValidatorAccountant is ValidatorRegistrator {
     /// accounting is valid and fuse isn't "blown". Returns false when fuse is blown.
     /// @dev This function could in theory be permission-less but lets allow only the Registrator (Defender Action) to call it
     /// for now.
+    /// @return accountingValid true if accounting was successful, false if fuse is blown
     /* solhint-enable max-line-length */
     function doAccounting()
         external
         onlyRegistrator
         whenNotPaused
         returns (bool accountingValid)
     {
+        accountingValid = _doAccounting();
+    }
+
+    function _doAccounting() internal returns (bool accountingValid) {
```

Comment:
I know this works for now, I am slightly worried that we will in future forget that internal _doAccounting needs to be in `whenNotPaused` context when doing the accounting and in `whenPaused` context when manually correcting the accounting.

What if we introduce a bool property to the internal `_doAccounting` function and be explicit whether a `_pause()` call should be made depending on who the caller of the transaction is. Smth like: 

```
function _doAccounting(bool callPause) internal...
...
  // pause and fail the accounting
  if (callPause) {
      _pause();
  }
  // fail the accounting
  return false;
....
```

Reply:
using a `pauseOnFail` param also saves as storage slot read. I'll make the change